source/FPSciApp.cpp

```diff
@@ -564,16 +569,22 @@ void FPSciApp::updateSession(const String& id, bool forceReload) {
 		hudTextures.set(element.filename, Texture::fromFile(System::findDataFile(element.filename)));
 	}
 
-	// Create a series of colored materials to choose from for target health
-	m_materials.clear();
-	for (int i = 0; i < m_MatTableSize; i++) {
-		float complete = (float)i / m_MatTableSize;
-		Color3 color = sessConfig->targetView.healthColors[0] * complete + sessConfig->targetView.healthColors[1] * (1.0f - complete);
-		UniversalMaterial::Specification materialSpecification;
-		materialSpecification.setLambertian(Texture::Specification(color));
-		materialSpecification.setEmissive(Texture::Specification(color * 0.7f));
-		materialSpecification.setGlossy(Texture::Specification(Color4(0.4f, 0.2f, 0.1f, 0.8f)));
-		m_materials.append(UniversalMaterial::create(materialSpecification));
+	// Update colored materials to choose from for target health
+	for (String id : sessConfig->getUniqueTargetIds()) {
+		Array<shared_ptr<UniversalMaterial>> targetMaterials;
+		shared_ptr<TargetConfig> tconfig = experimentConfig.getTargetConfigById(id);
+		for (int i = 0; i < m_MatTableSize; i++) {
+			float complete = (float)i / m_MatTableSize;
+			Color3 color;
+			if (notNull(tconfig) && tconfig->colors.length() > 0) { color = tconfig->colors[0] * complete + tconfig->colors[1] * (1.0f - complete); }
```

Comment:
Seems like this should be checking `tconfig->colors.length() >= 2` or `tconfig->colors.length() > 1` since it's using `tconfig->colors[1]`.